### PR TITLE
Configurable socket pooling

### PR DESCRIPTION
**Pull request by stowns**
When running jobs that require many parallel connections to facebook the default socket pool becomes a performance issue.  This should help with that.

### CODE REVIEW

File: fb.js [hunk 1]
```diff
@@ -275,11 +276,13 @@
                 }
                 uri = uri.substring(0, uri.length -1);
             };
-
+            
+            var pool = { maxSockets : Number(process.env.MAX_SOCKETS) || 5 };
```
Comment (prabirshrestha) [inline fb.js]: declare `var pool` at the top.

Comment (stowns) [inline fb.js]: done

File: fb.js [hunk 2]
```diff
@@ -275,11 +276,13 @@
                 }
                 uri = uri.substring(0, uri.length -1);
             };
-
+            
+            pool = { maxSockets : Number(process.env.MAX_SOCKETS) || 5 };
```
Comment (prabirshrestha) [inline fb.js]: I think it is first good to check from the options variable with `options('maxSockets')` and then fallback to env with default 5. also make sure to document it.

**Comment by stowns** [comment]
done

**Pull request merged**